pkgs/development/libraries/science/math/libamplsolver/default.nix

```diff
@@ -9,19 +9,23 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-LVmScuIvxmZzywPSBl9T9YcUBJP7UFAa3eWs9r4q3JM=";
   };
 
-  patches = [
-    # Debian provides a patch to build a shared library
-    (fetchpatch {
-      url = "https://sources.debian.org/data/main/liba/libamplsolver/0~20190702-2/debian/patches/fix-makefile-shared-lib.patch";
-      sha256 = "sha256-96qwj3fLugzbsfxguKMce13cUo7XGC4VUE7xKcJs42Y=";
-    })
-  ];
+  patches = if stdenv.isLinux
+  then
+    [
+      # Debian provides a patch to build a shared library
+      (fetchpatch {
+        url = "https://sources.debian.org/data/main/liba/libamplsolver/0~20190702-2/debian/patches/fix-makefile-shared-lib.patch";
+        sha256 = "sha256-96qwj3fLugzbsfxguKMce13cUo7XGC4VUE7xKcJs42Y=";
+      })
+    ]
+  else
+    [ ./libamplsolver-darwin.patch ];
```

Comment:
Please leave a quick comment here and I'm happy to merge. Thanks for adding support! :sparkles:

Reply:
I don't know enough C++ to do that, but if you know do please send a suggestion!

Reply:
You don't need to know c++ for this, only some conditional make. Should be pretty easy to fix.

Reply:
Determining OS from makefiles seems like a [bit of a mess](https://www.techtalk7.com/os-detecting-makefile/). Is there strong reason to do this instead of using the already robust OS detection built into nix?

Engineering our own homegrown omnibus patch that works for both mac and linux seems difficult to make robust (e.g. how will it work for cross-compilation? I have no clue), especially for a non-expert like me, and makes maintenance more difficult since we can't just track the debian patch.

Reply:
> Is there strong reason to do this instead of using the already robust OS detection built into nix?

Yes, those patches will definitely be updated if they no longer apply with an update. Right now the darwin patch will probably be forgotten to be checked.